Application/Published/Upgrade-AdminApp/Deploy-EdFiOdsAdminApp.ps1

```diff
@@ -0,0 +1,220 @@
+
+$AdminAppTemplateFile = "$TemplateFileDirectory\OdsAdminApp.Upgrade.json"
+$AdminAppTemplateParametersFile = "$TemplateFileDirectory\OdsAdminApp.parameters.Upgrade.json"
+
+function Deploy-AdminApp()
+{
+	if(-not $EncryptionKey)
+	{
+		$aes = [System.Security.Cryptography.Aes]::Create()
+		$aes.KeySize = 256
+		$aes.GenerateKey()
+		$EncryptionKey = [System.Convert]::ToBase64String($aes.Key)
+	}
+	
+	$deployParameters = New-Object -TypeName Hashtable
+	$deployParameters.Add("version", $AdminAppVersion)
+	$deployParameters.Add("edition", $Edition)
+	$deployParameters.Add("appInsightsLocation", $AppInsightLocation)
+	$deployParameters.Add("odsInstanceName", $ResourceGroupName)
+	$deployParameters.Add("sqlServerAdminLogin", $SQLServerUserName)
+	$deployParameters.Add("sqlServerAdminPassword", $SQLServerPassword)
+	$deployParameters.Add("productionApiUrl", $ProductionApiUrl)	
+	$deployParameters.Add("metadataCacheTimeOut", $CacheTimeOut)
+	$deployParameters.Add("adminAppNameToDeploy", $AdminAppName)
+	$deployParameters.Add("encryptionKey", $EncryptionKey)
+	
+	$templateFile = $AdminAppTemplateFile
+	$templateParametersFile = $AdminAppTemplateParametersFile
+
+	Try
+	{
+		Write-Host "Deploying ODS Admin App"		
+		$deploymentResult = New-AzureRmResourceGroupDeployment -DeploymentDebugLogLevel All -Name ((Get-ChildItem $templateFile).BaseName + '-' + ((Get-Date).ToUniversalTime()).ToString('MMdd-HHmm')) `
+									   -ResourceGroupName $ResourceGroupName `
+									   -TemplateFile $templateFile `
+									   -TemplateParameterFile $templateParametersFile `
+									   @deployParameters `
+									   -Force -Verbose -ErrorAction Stop
+	}
+	Catch
+	{
+		Write-Error $_.Exception.Message		
+	}
+
+	$adminAppUrl = $deploymentResult.Outputs.adminAppUrl.Value
+	Write-Success "Ed-Fi ODS Admin App accessible at $adminAppUrl"
+	
+	return $deploymentResult
+}
+
+function IsDBInstalled($pwd, [string]$database)
+{	 
+	$arguments = @{
+		ServerInstance = $SQLServerHostname
+		Username = $SQLServerUserName
+		Password = $pwd
+		Database = "master"
+		Query = "select 1 from sys.databases where name='$database'"
+		OutputSqlErrors = $true
+	}
+
+	$result = Invoke-Sqlcmd @arguments 
+	if (!$result) {				
+		Write-Error "Failed to connect to [$database] database on [$Server]" -ErrorAction Stop
+	} else {				   
+		Write-Success "[$database] Database exists in SQL Server [$Server]"
+	}
+}
+
+function Run-DbMigrations()
+{
+	$scriptPackageUri = "https://odsassets.blob.core.windows.net/public/adminapp/$Edition/$AdminAppVersion/Edfi.suite3.ods.adminapp.database.zip"
+	$destinationZipFile = "$PSScriptRoot/Database.zip"
+	$scriptFilesPath = "$PSScriptRoot/ScriptFiles"
+	Invoke-WebRequest -Uri $scriptPackageUri -OutFile $destinationZipFile
+	Expand-Archive $destinationZipFile -DestinationPath $scriptFilesPath -Force	
+	
+	$database = "EdFi_Admin"
+	$pwd = (New-Object PSCredential $SQLServerUserName, $SQLServerPassword).GetNetworkCredential().Password	
```

Comment:
`$SQLServerPassword` is already a plain string, so I don't think you need this line anymore.

Reply:
@stephenfuqua 
We are receiving as Secure string. So need this line to convert.

Reply:
You're right, I misread the function argument.